exporter/metrics.go

```diff
@@ -35,26 +119,39 @@ func makeRawMetric(name string, value interface{}) (prometheus.Metric, error) {
 		f = float64(v)
 	case int64:
 		f = float64(v)
+	case float32:
+		f = float64(v)
 	case float64:
 		f = v
-	case primitive.A, primitive.ObjectID: // TODO: Fix this. returning nil, nil just to make tests pass
-		return nil, nil
-
 	case primitive.DateTime:
 		f = float64(v)
-	case primitive.Timestamp:
-		return nil, nil
-
-	case string:
+	case primitive.A, primitive.ObjectID, primitive.Timestamp, string:
 		return nil, nil
-
 	default:
 		return nil, fmt.Errorf("makeRawMetric: unhandled type %T", v)
 	}
 
-	fqName := makeRawMetricName(name)
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
 	help := "TODO"
```

Comment:
I think we should synthesise help test even now. For example, use original metric name.